operator/pkg/gateway-api/gatewayclass.go

```diff
@@ -32,12 +45,47 @@ func newGatewayClassReconciler(mgr ctrl.Manager, logger *slog.Logger) *gatewayCl
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *gatewayClassReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	for indexName, indexerFunc := range map[string]client.IndexerFunc{
+		gatewayClassConfigMapIndexName: referencedConfig,
+	} {
+		if err := mgr.GetFieldIndexer().IndexField(context.Background(), &gatewayv1.GatewayClass{}, indexName, indexerFunc); err != nil {
+			return fmt.Errorf("failed to setup field indexer %q: %w", indexName, err)
+		}
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gatewayv1.GatewayClass{},
 			builder.WithPredicates(predicate.NewPredicateFuncs(matchesControllerName(controllerName)))).
+		Watches(&v2alpha1.CiliumGatewayClassConfig{}, r.enqueueRequestForConfigMap()).
```

Comment:
```suggestion
		Watches(&v2alpha1.CiliumGatewayClassConfig{}, r.enqueueRequestForCGCC()).
```
Or maybe the full name? `enqueueRequestForCiliumGatewayClassConfig()`?

Reply:
thanks